service/history/shard/context_test.go

```diff
@@ -203,6 +243,77 @@ func (s *contextTestSuite) TestReplicateFailoverMarkersSuccess() {
 	s.NoError(err)
 }
 
+func (s *contextTestSuite) TestCreateWorkflowExecution_Succeeds() {
+	ctx := context.Background()
+	request := &persistence.CreateWorkflowExecutionRequest{
+		DomainName: testDomain,
+		NewWorkflowSnapshot: persistence.WorkflowSnapshot{
+			ExecutionInfo: &persistence.WorkflowExecutionInfo{
+				DomainID:   testDomainID,
+				WorkflowID: testWorkflowID,
+			},
+		},
+	}
+
+	domainCacheEntry := cache.NewLocalDomainCacheEntryForTest(
+		&persistence.DomainInfo{ID: testDomainID},
+		&persistence.DomainConfig{Retention: 7},
+		testCluster,
+	)
+	s.mockResource.DomainCache.EXPECT().GetDomainByID(testDomainID).Return(domainCacheEntry, nil)
+
+	persistenceResponse := &persistence.CreateWorkflowExecutionResponse{}
+	s.mockResource.ExecutionMgr.On("CreateWorkflowExecution", ctx, mock.Anything).Once().Return(persistenceResponse, nil)
+
+	resp, err := s.context.CreateWorkflowExecution(ctx, request)
+	s.NoError(err)
+	s.NotNil(resp)
```

Comment:
Would it be possible to check something more than just not nil?

Reply:
Ouch. Completely missed your question.
Unfortunately, didn't find any meaningful return value there. resp' content is not really having anything related to test. It's more to check mocks.